client/List.jsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+class Item extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = {isCompleted: false};
+        this.handleClick = this.handleClick.bind(this);
+    }
+    handleClick(){
+        this.setState({isCompleted: !this.state.isCompleted});
+    }
+    render(){
+        const {name} = this.props;
+        const itemWithStatus = this.state.isCompleted ? <strike>{name}</strike> : name; 
+        return (
+        <span onClick={this.handleClick}>
+            {itemWithStatus}
+        </span>
+        );
+    }
+};
+
+const List = (props) => {
+  const { items } = props;
+
+  return (
+    <ul>{items.map((item, index) => <li key={`item_${index}`}><Item name={item}/></li>)} </ul>
+  );
+};
+
+export class DynamicList extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = { items: props.items };
+        this.handleAddItem = this.handleAddItem.bind(this);
+    }
+
+    handleAddItem(e) {
+        if (e.key == "Enter") {
+            const newItem = e.target.value.length > 0 ? e.target.value : null;
+            if (newItem) {
+                e.target.value = ""
+                this.setState({items: [...this.state.items, newItem]});
+            }
+        }
+        
+    }
+    render() {
+        return (
+            <div>
```

Comment:
Just as an FYI, you don't need wrapper elements in React 16.  React 16 allows you to create adjacent elements using an array.  Something like:
```
return [
    <h1>Some heading</h1>,
    <h2>Some subheading</h2>
];
```

Reply:
idk why but watch craps out when I use the array notation

Reply:
@hshercha that's weird, do you have a console output?

Reply:
And also can you paste the code?

Reply:
nvm it was dumb. I didnt realize I had to add the comma-separated values